onedal/basic_statistics/tests/test_incremental_basic_statistics.py

```diff
@@ -0,0 +1,249 @@
+        weights = weights.astype(dtype=dtype)
+        weights_split = np.array_split(weights, num_batches)
+    incbs = IncrementalBasicStatistics(result_options=result_option)
+
+    for i in range(num_batches):
+        if weighted:
+            incbs.partial_fit(data_split[i], weights_split[i], queue=queue)
+        else:
+            incbs.partial_fit(data_split[i], queue=queue)
+    result = incbs.finalize_fit()
+
+    res = getattr(result, result_option)
+    if weighted:
+        weighted_data = np.diag(weights) @ data
+        gtr = function(weighted_data)
+    else:
+        gtr = function(data)
+
+    tol = fp32tol if res.dtype == np.float32 else fp64tol
+    assert_allclose(gtr, res, rtol=tol, atol=tol)
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("num_batches", [2, 10])
+@pytest.mark.parametrize("row_count", [100, 1000])
+@pytest.mark.parametrize("column_count", [10, 100])
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_multiple_options_on_random_data(
+    queue, num_batches, row_count, column_count, weighted, dtype
+):
+    seed = 42
+    gen = np.random.default_rng(seed)
+    data = gen.uniform(low=-0.3, high=+0.7, size=(row_count, column_count))
+    data = data.astype(dtype=dtype)
+    data_split = np.array_split(data, num_batches)
+    if weighted:
+        weights = gen.uniform(low=-0.5, high=+1.0, size=row_count)
+        weights = weights.astype(dtype=dtype)
+        weights_split = np.array_split(weights, num_batches)
+    incbs = IncrementalBasicStatistics(result_options=["mean", "max", "sum"])
+
+    for i in range(num_batches):
+        if weighted:
+            incbs.partial_fit(data_split[i], weights_split[i], queue=queue)
+        else:
+            incbs.partial_fit(data_split[i], queue=queue)
+    result = incbs.finalize_fit()
+
+    res_mean, res_max, res_sum = result.mean, result.max, result.sum
+    if weighted:
+        weighted_data = np.diag(weights) @ data
+        gtr_mean, gtr_max, gtr_sum = (
+            expected_mean(weighted_data),
+            expected_max(weighted_data),
+            expected_sum(weighted_data),
+        )
+    else:
+        gtr_mean, gtr_max, gtr_sum = (
+            expected_mean(data),
+            expected_max(data),
+            expected_sum(data),
+        )
+
+    tol = 1e-5 if res_mean.dtype == np.float32 else 1e-7
+    assert_allclose(gtr_mean, res_mean, rtol=tol, atol=tol)
+    assert_allclose(gtr_max, res_max, rtol=tol, atol=tol)
+    assert_allclose(gtr_sum, res_sum, rtol=tol, atol=tol)
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("num_batches", [2, 10])
+@pytest.mark.parametrize("row_count", [100, 1000])
+@pytest.mark.parametrize("column_count", [10, 100])
+@pytest.mark.parametrize("weighted", [True, False])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_all_option_on_random_data(
```

Comment:
```suggestion
def test_all_options_on_random_data(
```

Reply:
it should be read as 'test "all" option', so the current name is ok as well. but if you insist I can change.

Reply:
I see, in that case can keep it as option